h2o-core/src/main/java/water/Scope.java

```diff
@@ -64,13 +64,18 @@ static void track_internal( Key k ) {
     track_impl(scope, k);
   }
 
-  static public <T extends Keyed> Keyed<T> track_generic(Keyed<T> keyed) {
+  static public <T extends Keyed> T track(T keyed) {
     Scope scope = _scope.get();                   // Pay the price of T.L.S. lookup
     assert scope != null;
     track_impl(scope, keyed._key);
     return keyed;
   }
 
+  @Deprecated
+  static public <T extends Keyed> T track_generic(T keyed) {
```

Comment:
This function is only currently used in w2v algo. If you want to rename the function, you don't have to keep the old one.